### ISSUE TITLE
Provisioning a Smart Wallet for an address multiple times breaks the address

### ISSUE THREAD

**Issue by arirubinstein** [Agoric/Agoric#6295]
If I provision a smart wallet once, everything works well.  If I provision the same address twice, the wallet seems to get bugged.
I'm using the instagoric faucet provision-one interface to reproduce this however I suspect the Pay 10 BLD for a wallet may be vulnerable to this as well.

Expected: Provisioning an address many times should not break the wallet

**Comment by dckc** [comment]
I'm struggling to reproduce the problem.

I provisioned `agoric18mshnndqxzpy2falwn9rn69qpq7598aplvdyuh` twice on ollinet; the result was that I got 2 x 25.5 IST.

<details>
<summary>screenshot, transaction details</summary>

![image](https://user-images.githubusercontent.com/150986/191614237-390ccf53-9c67-4757-9789-29c9e1f08aaf.png)


txhash | timestamp | height | type | detail | To
-- | -- | -- | -- | -- | --
35CC14BF788F23C612EAD5869AE0ED24156CE157112F080CAFB4948F28AF37E1 | 2022-09-21 16:10 | 30239 | send | 25 IBC/USD..234, 25 IST | agoric18mshnndqxzpy2falwn9rn69qpq7598aplvdyuh
F6C1AF27199A5D89CD458E7332CD87AFB215EB285ABA9E79191197387AAFFB8C | 2022-09-21 16:10 | 30240 | provision | SMART_WALLET | agoric18mshnndqxzpy2falwn9rn69qpq7598aplvdyuh
D2B52FBA3DC7D19D46D793CD022ABE9739EFCB70B7FD39B693F9DD2D391C7EAC | 2022-09-21 16:10 | 30246 | send | 25 IBC/USD..234, 25 IST | agoric18mshnndqxzpy2falwn9rn69qpq7598aplvdyuh
46341AB5538BF94C5F5921DE12FBB389E15AAFD4881938CE1F799B65F3121FD6 | 2022-09-21 16:11 | 30247 | provision | SMART_WALLET | agoric18mshnndqxzpy2falwn9rn69qpq7598aplvdyuh

</details>

**Comment by arirubinstein** [comment]
hmm,  is it possible to see if `agoric1qpms2h23zct7uumdz3ve7jsw90jt8r03cgq0nx` is in a bad state on ollinet?

**Comment by turadg** [comment]
Yep, that one's in a bad state. `AGORIC_NET=ollinet agoric wallet show --from agoric1qpms2h23zct7uumdz3ve7jsw90jt8r03cgq0nx` shows,
```
got state {
  brands: Map(0) {},
  offerStatuses: Map(1) {
    1663650870534 => {
      id: 1663650870534,
      invitationSpec: [Object],
      proposal: [Object],
      numWantsSatisfied: 1,
      payouts: [Object]
    }
  },
  balances: Map(2) {
    { boardId: 'board0188', iface: 'Alleged: IST brand' } => { brand: [Object], value: 25249800n },
    { boardId: 'board0223', iface: 'Alleged: AUSD brand' } => { brand: [Object], value: 24999400n }
  }
}
```

The `brands` map should have at least `board0188` and `board0223` that are in the purse balances.

**Comment by dckc** [comment]
I was starting to look into slogs for that address, but meeting time came up, and now it looks like the cluster was restarted, and I didn't manage to save the slogs. Darn.

**Comment by dckc** [comment]
I think I see the/a bug now... we use the `provide` pattern for the smart wallet itself, but not for `  makeMyAddressNameAdminKit`:

https://github.com/Agoric/agoric-sdk/blob/f73c2a7b8dc97709d6dd9c60c55913bb916983fe/packages/vats/src/provisionPool.js#L73-L74

I had some code to avoid doing that twice, but it resulted in a deadlock when working on econ committee voting stuff (remember, @Chris-Hibbert ?) , so I/we took it out.

**Comment by dckc** [comment]
closed in #6307
